components/compositing/constellation.rs

```diff
@@ -638,17 +638,11 @@ impl<LTF: LayoutTaskFactory, STF: ScriptTaskFactory> Constellation<LTF, STF> {
 
         self.close_pipeline(pipeline_id, ExitPipelineMode::Force);
 
-        loop {
```

Comment:
Unsure what to do here @Manishearth. I input your suggestion the for loop and came to the same conclusion you did.